modules/apps/frontend-js/frontend-js-spa-web/src/main/resources/META-INF/resources/screen/HtmlScreen.js

```diff
@@ -157,6 +157,32 @@ class HtmlScreen extends RequestScreen {
 		);
 	}
 
+	/**
+	 * @Override
+	 */
+	preloadResources(surfaces) {
+		const tracked = this.virtualQuerySelectorAll_(
+			HtmlScreen.selectors.styles
```

Comment:
Any reason for the special handling of `selectors` and `permanentResourcesInDoc`? Why are they not handled as all other properties of the `HtmlScreen` class, and accessed with `this`? This is not related to PR, it's ok to follow existing solution, I'm just curious.

Reply:
I have no idea, I copy pasted that code from the function that finally applies the styles. Meaning I made sure I preload all the styles the other function will be using.

Perhaps I should refactor this to centralize that code so that it remains the same forever, for both uses... :thinking:

Reply:
Couldn't make the refactor :shrug: -> https://github.com/liferay-frontend/liferay-portal/pull/4338/commits/740aef501f24359f417353672011a46e94178224